### ISSUE TITLE
about noise

### ISSUE THREAD

**Issue by jnoylinc** [prote376/DBPI-BlindSR#3]
Thanks for your work，I want to ask your a question about the code:
g_updown = self.G(torch.clamp(g_up.detach() + 1e-1 * torch.randn_like(g_up), -1, 1))
1e-1 * torch.randn_like(g_up) means the noise?
I get blur result when I use g_updown = self.G(g_up.detach()), and the loss is lower than before.
Can you give me an explanation?

**Comment by prote376** [comment]
Yes. That is a kind of data augmentation by adding noise.
The data augmentation can prevent overfitting and make learning better.

**Comment by jnoylinc** [comment]
Thanks for your reply! 
what can I do if I want to improve the sharpness of the result?
why do you add noise in g_updown? I find that if I increase the weight of noise, the sharpness of the result is improved

**Comment by jnoylinc** [comment]
Have you tried the ssim vgg loss?

**Comment by prote376** [comment]
> Thanks for your reply!
> what can I do if I want to improve the sharpness of the result?
> why do you add noise in g_updown? I find that if I increase the weight of noise, the sharpness of the result is improved

As I mentioned above, I added noise to prevent overfitting.
It could be more sharp.
Since the downscaling network blurs the upscaled image more to remove the high-level noise, the SR network is trained to generate more sharp images to revert the highly blurred image. 
But, it could generate over-sharpen images, too.

**Comment by prote376** [comment]
> Have you tried the ssim vgg loss?

Not yet. I wanted to keep the proposed method simple. You could try to improve the performance.
